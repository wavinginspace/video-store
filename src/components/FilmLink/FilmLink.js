import React from 'react';
import { Link } from 'react-router-dom';
import './FilmLink.scss';

function FilmLink(props) {
  const { film } = props;
  return (
    <Link to={`/films/${film.id}`} className="film-link" title={film}>
      <li className="FilmLink" title={film}>
        {film}
      </li>
    </Link>
  );
}

export default FilmLink;
