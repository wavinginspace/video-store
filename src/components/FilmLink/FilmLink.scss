@import '../../styles/_breakpoints.scss';
@import '../../styles/_variables.scss';

.film-link {
  text-decoration: none;
  list-style-type: none;
  color: var(--main);
}

.FilmLink {
  margin: 1rem auto;
}

.FilmLink:hover {
  color: blue;
}
