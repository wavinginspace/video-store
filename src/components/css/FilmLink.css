.film-link {
  text-decoration: none;
  list-style-type: none;
  color: black;
}

.FilmLink {
  margin: 1rem auto;
}

.FilmLink:hover {
  color: blue;
}
/*# sourceMappingURL=FilmLink.css.map */