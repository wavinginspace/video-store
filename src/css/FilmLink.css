:root {
  --main: var(--main);
  --nav: #499acc;
  --headergrad: linear-gradient(0deg, #ff7676, #fffd8f);
  --light: #fffd8f;
  --headerog: #ff7676;
}

.film-link {
  text-decoration: none;
  list-style-type: none;
  color: var(--main);
}

.FilmLink {
  margin: 1rem auto;
}

.FilmLink:hover {
  color: blue;
}
/*# sourceMappingURL=FilmLink.css.map */