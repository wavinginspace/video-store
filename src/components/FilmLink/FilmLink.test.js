import React from 'react';
import ReactDOM from 'react-dom';
import { MemoryRouter as Router} from 'react-router-dom';
import FilmLink from './FilmLink';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  // first create a DOM element to render the component into
  const div = document.createElement('div');

  // render the component, this is the actual test, if something is wrong it will fail here
  ReactDOM.render(<Router><FilmLink film='test' /></Router>, div);

  // clean up code
  ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected', () => {
  const tree = renderer.create(<Router><FilmLink film='test'/></Router>).toJSON();
  expect(tree).toMatchSnapshot();
});
