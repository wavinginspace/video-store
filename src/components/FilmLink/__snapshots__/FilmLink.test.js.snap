// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<a
  className="film-link"
  href="/films/undefined"
  onClick={[Function]}
>
  <li
    className="FilmLink"
  />
</a>
`;
